MapboxGeocoder/MBGeocoder.swift

```diff
@@ -346,6 +356,114 @@ open class Geocoder: NSObject {
         components.queryItems = params
         return components.url!
     }
+    
+    /**
+     Send a keydown event
+    */
+    func initKeydown(options: GeocodeOptions){
+        if (options.queries.count > 1) {
+            // don't record batch geocodes
+            return
+        }
+        let query = options.queries[0]
+        if (isReverseGeocode(testStr: query)){
+            // don't record reverse geocodes
+            return
+        }
+        if (query == self.lastQuery){
+            // don't duplicate
+            return
+        }
+        if (query == ""){
+            // don't record empty queries
+            return
+        }
+        
+        let lastAction = stringDiff(a: query, b:self.lastQuery)
+        let now =  NSDate().timeIntervalSince1970
```

Comment:
Nit: `Date` is more Swift-like than `NSDate`, which is a holdover from Objective-C.